src/librustc_middle/ty/mod.rs

```diff
@@ -2399,7 +2399,11 @@ impl<'tcx> AdtDef {
                 None
             }
             Err(ErrorHandled::TooGeneric) => {
-                span_bug!(tcx.def_span(expr_did), "enum discriminant depends on generic arguments",)
+                tcx.sess.delay_span_bug(
+                    tcx.def_span(expr_did),
+                    "enum discriminant depends on generic arguments",
+                );
+                None
```

Comment:
This *must be* the actual change here, and it looks a lot like the "ty: use `delay_span_bug` in `ty::AdtDef::eval_explicit_discr`." change from #70825.

Reply:
It is.